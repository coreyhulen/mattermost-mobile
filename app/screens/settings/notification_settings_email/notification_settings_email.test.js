// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
// See License.txt for license information.

import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({adapter: new Adapter()});

import NotificationSettingsEmail from './notification_settings_email.js';

// jest.useFakeTimers();

jest.mock('app/utils/theme', () => {
    const original = require.requireActual('app/utils/theme');
    return {
        ...original,
        changeOpacity: jest.fn(),
    };
});

describe('NotificationSettingsEmail', () => {
    const baseProps = {
        currentUser: {id: 'user_id', notify_props: {email: 'true'}},
        emailInterval: '30',
        enableEmailBatching: false,
        navigator: {setOnNavigatorEvent: () => {}}, // eslint-disable-line no-empty-function
        onBack: () => {}, // eslint-disable-line no-empty-function
        sendEmailNotifications: true,
        siteName: 'Mattermost',
        theme: {
            centerChannelBg: '#aaa',
            centerChannelColor: '#aaa',
        },
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <NotificationSettingsEmail {...baseProps}/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should call saveUserNotifyProps on onNavigatorEvent', () => {
        const wrapper = shallow(
            <NotificationSettingsEmail {...baseProps}/>
        );

        const instance = wrapper.instance();
        instance.saveUserNotifyProps = jest.fn();
        instance.onNavigatorEvent({type: 'ScreenChangedEvent', id: 'willDisappear'});

        expect(instance.saveUserNotifyProps).toHaveBeenCalledTimes(1);
    });

    test('should macth state on setEmailNotifications', () => {
        const wrapper = shallow(
            <NotificationSettingsEmail {...baseProps}/>
        );

        wrapper.setState({email: 'false', interval: '0'});
        wrapper.instance().setEmailNotifications('30');
        expect(wrapper.state({email: 'true', interval: '30'}));

        wrapper.instance().setEmailNotifications('0');
        expect(wrapper.state({email: 'false', interval: '0'}));

        wrapper.instance().setEmailNotifications('3600');
        expect(wrapper.state({email: 'true', interval: '3600'}));
    });

    test('should call props.onBack on saveUserNotifyProps', () => {
        const props = {...baseProps, onBack: jest.fn()};
        const wrapper = shallow(
            <NotificationSettingsEmail {...props}/>
        );

        wrapper.setState({email: 'true', interval: '3600'});
        wrapper.instance().saveUserNotifyProps();
        expect(props.onBack).toHaveBeenCalledTimes(1);
        expect(props.onBack).toBeCalledWith({email: 'true', interval: '3600', user_id: 'user_id'});
    });

    test('should match snapshot, renderEmailSection', () => {
        const wrapper = shallow(
            <NotificationSettingsEmail {...baseProps}/>
        );

        expect(wrapper.instance().renderEmailSection()).toMatchSnapshot();
    });
});