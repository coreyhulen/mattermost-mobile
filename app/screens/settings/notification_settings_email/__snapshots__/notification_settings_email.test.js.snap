// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationSettingsEmail should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotificationSettingsEmail
    currentUserId="current_user_id"
    emailInterval="30"
    enableEmailBatching={false}
    navigator={
      Object {
        "setOnNavigatorEvent": [Function],
      }
    }
    savePreferences={[Function]}
    sendEmailNotifications={true}
    siteName="Mattermost"
    theme={
      Object {
        "centerChannelBg": "#aaa",
        "centerChannelColor": "#aaa",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Connect(StatusBar) />,
        <ScrollView
          alwaysBounceVertical={false}
          contentContainerStyle={
            Object {
              "paddingVertical": 35,
            }
          }
          style={
            Object {
              "backgroundColor": undefined,
              "flex": 1,
            }
          }
        >
          <section
            disableFooter={false}
            footerDefaultMessage="Email notifications are sent for mentions and direct messages when you are offline or away from {siteName} for more than 5 minutes."
            footerId="user.settings.notifications.emailInfo"
            footerValues={
              Object {
                "siteName": "Mattermost",
              }
            }
            headerDefaultMessage="SEND EMAIL NOTIFICATIONS"
            headerId="mobile.notification_settings.email.send"
            theme={
              Object {
                "centerChannelBg": "#aaa",
                "centerChannelColor": "#aaa",
              }
            }
          >
            <Component>
              <sectionItem
                action={[Function]}
                actionType="select"
                actionValue="30"
                label={
                  <InjectIntl(FormattedText)
                    defaultMessage="Immediately"
                    id="user.settings.notifications.email.immediately"
                  />
                }
                selected={true}
                theme={
                  Object {
                    "centerChannelBg": "#aaa",
                    "centerChannelColor": "#aaa",
                  }
                }
              />
              <Component
                style={
                  Object {
                    "backgroundColor": undefined,
                    "flex": 1,
                    "height": 1,
                    "marginLeft": 15,
                  }
                }
              />
              <sectionItem
                action={[Function]}
                actionType="select"
                actionValue="0"
                label={
                  <InjectIntl(FormattedText)
                    defaultMessage="Never"
                    id="user.settings.notifications.email.never"
                  />
                }
                selected={false}
                theme={
                  Object {
                    "centerChannelBg": "#aaa",
                    "centerChannelColor": "#aaa",
                  }
                }
              />
            </Component>
          </section>
        </ScrollView>,
      ],
      "style": Object {
        "backgroundColor": "#aaa",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alwaysBounceVertical": false,
          "children": <section
            disableFooter={false}
            footerDefaultMessage="Email notifications are sent for mentions and direct messages when you are offline or away from {siteName} for more than 5 minutes."
            footerId="user.settings.notifications.emailInfo"
            footerValues={
              Object {
                "siteName": "Mattermost",
              }
            }
            headerDefaultMessage="SEND EMAIL NOTIFICATIONS"
            headerId="mobile.notification_settings.email.send"
            theme={
              Object {
                "centerChannelBg": "#aaa",
                "centerChannelColor": "#aaa",
              }
            }
          >
            <Component>
              <sectionItem
                action={[Function]}
                actionType="select"
                actionValue="30"
                label={
                  <InjectIntl(FormattedText)
                    defaultMessage="Immediately"
                    id="user.settings.notifications.email.immediately"
                  />
                }
                selected={true}
                theme={
                  Object {
                    "centerChannelBg": "#aaa",
                    "centerChannelColor": "#aaa",
                  }
                }
              />
              <Component
                style={
                  Object {
                    "backgroundColor": undefined,
                    "flex": 1,
                    "height": 1,
                    "marginLeft": 15,
                  }
                }
              />
              <sectionItem
                action={[Function]}
                actionType="select"
                actionValue="0"
                label={
                  <InjectIntl(FormattedText)
                    defaultMessage="Never"
                    id="user.settings.notifications.email.never"
                  />
                }
                selected={false}
                theme={
                  Object {
                    "centerChannelBg": "#aaa",
                    "centerChannelColor": "#aaa",
                  }
                }
              />
            </Component>
          </section>,
          "contentContainerStyle": Object {
            "paddingVertical": 35,
          },
          "style": Object {
            "backgroundColor": undefined,
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Component>
                <sectionItem
                  action={[Function]}
                  actionType="select"
                  actionValue="30"
                  label={
                    <InjectIntl(FormattedText)
                      defaultMessage="Immediately"
                      id="user.settings.notifications.email.immediately"
                    />
                  }
                  selected={true}
                  theme={
                    Object {
                      "centerChannelBg": "#aaa",
                      "centerChannelColor": "#aaa",
                    }
                  }
                />
                <Component
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "flex": 1,
                      "height": 1,
                      "marginLeft": 15,
                    }
                  }
                />
                <sectionItem
                  action={[Function]}
                  actionType="select"
                  actionValue="0"
                  label={
                    <InjectIntl(FormattedText)
                      defaultMessage="Never"
                      id="user.settings.notifications.email.never"
                    />
                  }
                  selected={false}
                  theme={
                    Object {
                      "centerChannelBg": "#aaa",
                      "centerChannelColor": "#aaa",
                    }
                  }
                />
              </Component>,
              false,
            ],
            "disableFooter": false,
            "footerDefaultMessage": "Email notifications are sent for mentions and direct messages when you are offline or away from {siteName} for more than 5 minutes.",
            "footerId": "user.settings.notifications.emailInfo",
            "footerValues": Object {
              "siteName": "Mattermost",
            },
            "headerDefaultMessage": "SEND EMAIL NOTIFICATIONS",
            "headerId": "mobile.notification_settings.email.send",
            "theme": Object {
              "centerChannelBg": "#aaa",
              "centerChannelColor": "#aaa",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <sectionItem
                    action={[Function]}
                    actionType="select"
                    actionValue="30"
                    label={
                      <InjectIntl(FormattedText)
                        defaultMessage="Immediately"
                        id="user.settings.notifications.email.immediately"
                      />
                    }
                    selected={true}
                    theme={
                      Object {
                        "centerChannelBg": "#aaa",
                        "centerChannelColor": "#aaa",
                      }
                    }
                  />,
                  <Component
                    style={
                      Object {
                        "backgroundColor": undefined,
                        "flex": 1,
                        "height": 1,
                        "marginLeft": 15,
                      }
                    }
                  />,
                  false,
                  <sectionItem
                    action={[Function]}
                    actionType="select"
                    actionValue="0"
                    label={
                      <InjectIntl(FormattedText)
                        defaultMessage="Never"
                        id="user.settings.notifications.email.never"
                      />
                    }
                    selected={false}
                    theme={
                      Object {
                        "centerChannelBg": "#aaa",
                        "centerChannelColor": "#aaa",
                      }
                    }
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "action": [Function],
                    "actionType": "select",
                    "actionValue": "30",
                    "label": <InjectIntl(FormattedText)
                      defaultMessage="Immediately"
                      id="user.settings.notifications.email.immediately"
                    />,
                    "selected": true,
                    "theme": Object {
                      "centerChannelBg": "#aaa",
                      "centerChannelColor": "#aaa",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "style": Object {
                      "backgroundColor": undefined,
                      "flex": 1,
                      "height": 1,
                      "marginLeft": 15,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                false,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "action": [Function],
                    "actionType": "select",
                    "actionValue": "0",
                    "label": <InjectIntl(FormattedText)
                      defaultMessage="Never"
                      id="user.settings.notifications.email.never"
                    />,
                    "selected": false,
                    "theme": Object {
                      "centerChannelBg": "#aaa",
                      "centerChannelColor": "#aaa",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            false,
          ],
          "type": [Function],
        },
        "type": [MockFunction],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Connect(StatusBar) />,
          <ScrollView
            alwaysBounceVertical={false}
            contentContainerStyle={
              Object {
                "paddingVertical": 35,
              }
            }
            style={
              Object {
                "backgroundColor": undefined,
                "flex": 1,
              }
            }
          >
            <section
              disableFooter={false}
              footerDefaultMessage="Email notifications are sent for mentions and direct messages when you are offline or away from {siteName} for more than 5 minutes."
              footerId="user.settings.notifications.emailInfo"
              footerValues={
                Object {
                  "siteName": "Mattermost",
                }
              }
              headerDefaultMessage="SEND EMAIL NOTIFICATIONS"
              headerId="mobile.notification_settings.email.send"
              theme={
                Object {
                  "centerChannelBg": "#aaa",
                  "centerChannelColor": "#aaa",
                }
              }
            >
              <Component>
                <sectionItem
                  action={[Function]}
                  actionType="select"
                  actionValue="30"
                  label={
                    <InjectIntl(FormattedText)
                      defaultMessage="Immediately"
                      id="user.settings.notifications.email.immediately"
                    />
                  }
                  selected={true}
                  theme={
                    Object {
                      "centerChannelBg": "#aaa",
                      "centerChannelColor": "#aaa",
                    }
                  }
                />
                <Component
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "flex": 1,
                      "height": 1,
                      "marginLeft": 15,
                    }
                  }
                />
                <sectionItem
                  action={[Function]}
                  actionType="select"
                  actionValue="0"
                  label={
                    <InjectIntl(FormattedText)
                      defaultMessage="Never"
                      id="user.settings.notifications.email.never"
                    />
                  }
                  selected={false}
                  theme={
                    Object {
                      "centerChannelBg": "#aaa",
                      "centerChannelColor": "#aaa",
                    }
                  }
                />
              </Component>
            </section>
          </ScrollView>,
        ],
        "style": Object {
          "backgroundColor": "#aaa",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alwaysBounceVertical": false,
            "children": <section
              disableFooter={false}
              footerDefaultMessage="Email notifications are sent for mentions and direct messages when you are offline or away from {siteName} for more than 5 minutes."
              footerId="user.settings.notifications.emailInfo"
              footerValues={
                Object {
                  "siteName": "Mattermost",
                }
              }
              headerDefaultMessage="SEND EMAIL NOTIFICATIONS"
              headerId="mobile.notification_settings.email.send"
              theme={
                Object {
                  "centerChannelBg": "#aaa",
                  "centerChannelColor": "#aaa",
                }
              }
            >
              <Component>
                <sectionItem
                  action={[Function]}
                  actionType="select"
                  actionValue="30"
                  label={
                    <InjectIntl(FormattedText)
                      defaultMessage="Immediately"
                      id="user.settings.notifications.email.immediately"
                    />
                  }
                  selected={true}
                  theme={
                    Object {
                      "centerChannelBg": "#aaa",
                      "centerChannelColor": "#aaa",
                    }
                  }
                />
                <Component
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "flex": 1,
                      "height": 1,
                      "marginLeft": 15,
                    }
                  }
                />
                <sectionItem
                  action={[Function]}
                  actionType="select"
                  actionValue="0"
                  label={
                    <InjectIntl(FormattedText)
                      defaultMessage="Never"
                      id="user.settings.notifications.email.never"
                    />
                  }
                  selected={false}
                  theme={
                    Object {
                      "centerChannelBg": "#aaa",
                      "centerChannelColor": "#aaa",
                    }
                  }
                />
              </Component>
            </section>,
            "contentContainerStyle": Object {
              "paddingVertical": 35,
            },
            "style": Object {
              "backgroundColor": undefined,
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Component>
                  <sectionItem
                    action={[Function]}
                    actionType="select"
                    actionValue="30"
                    label={
                      <InjectIntl(FormattedText)
                        defaultMessage="Immediately"
                        id="user.settings.notifications.email.immediately"
                      />
                    }
                    selected={true}
                    theme={
                      Object {
                        "centerChannelBg": "#aaa",
                        "centerChannelColor": "#aaa",
                      }
                    }
                  />
                  <Component
                    style={
                      Object {
                        "backgroundColor": undefined,
                        "flex": 1,
                        "height": 1,
                        "marginLeft": 15,
                      }
                    }
                  />
                  <sectionItem
                    action={[Function]}
                    actionType="select"
                    actionValue="0"
                    label={
                      <InjectIntl(FormattedText)
                        defaultMessage="Never"
                        id="user.settings.notifications.email.never"
                      />
                    }
                    selected={false}
                    theme={
                      Object {
                        "centerChannelBg": "#aaa",
                        "centerChannelColor": "#aaa",
                      }
                    }
                  />
                </Component>,
                false,
              ],
              "disableFooter": false,
              "footerDefaultMessage": "Email notifications are sent for mentions and direct messages when you are offline or away from {siteName} for more than 5 minutes.",
              "footerId": "user.settings.notifications.emailInfo",
              "footerValues": Object {
                "siteName": "Mattermost",
              },
              "headerDefaultMessage": "SEND EMAIL NOTIFICATIONS",
              "headerId": "mobile.notification_settings.email.send",
              "theme": Object {
                "centerChannelBg": "#aaa",
                "centerChannelColor": "#aaa",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <sectionItem
                      action={[Function]}
                      actionType="select"
                      actionValue="30"
                      label={
                        <InjectIntl(FormattedText)
                          defaultMessage="Immediately"
                          id="user.settings.notifications.email.immediately"
                        />
                      }
                      selected={true}
                      theme={
                        Object {
                          "centerChannelBg": "#aaa",
                          "centerChannelColor": "#aaa",
                        }
                      }
                    />,
                    <Component
                      style={
                        Object {
                          "backgroundColor": undefined,
                          "flex": 1,
                          "height": 1,
                          "marginLeft": 15,
                        }
                      }
                    />,
                    false,
                    <sectionItem
                      action={[Function]}
                      actionType="select"
                      actionValue="0"
                      label={
                        <InjectIntl(FormattedText)
                          defaultMessage="Never"
                          id="user.settings.notifications.email.never"
                        />
                      }
                      selected={false}
                      theme={
                        Object {
                          "centerChannelBg": "#aaa",
                          "centerChannelColor": "#aaa",
                        }
                      }
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "action": [Function],
                      "actionType": "select",
                      "actionValue": "30",
                      "label": <InjectIntl(FormattedText)
                        defaultMessage="Immediately"
                        id="user.settings.notifications.email.immediately"
                      />,
                      "selected": true,
                      "theme": Object {
                        "centerChannelBg": "#aaa",
                        "centerChannelColor": "#aaa",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "style": Object {
                        "backgroundColor": undefined,
                        "flex": 1,
                        "height": 1,
                        "marginLeft": 15,
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  false,
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "action": [Function],
                      "actionType": "select",
                      "actionValue": "0",
                      "label": <InjectIntl(FormattedText)
                        defaultMessage="Never"
                        id="user.settings.notifications.email.never"
                      />,
                      "selected": false,
                      "theme": Object {
                        "centerChannelBg": "#aaa",
                        "centerChannelColor": "#aaa",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              false,
            ],
            "type": [Function],
          },
          "type": [MockFunction],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`NotificationSettingsEmail should match snapshot, renderEmailSection 1`] = `
<section
  disableFooter={false}
  footerDefaultMessage="Email notifications are sent for mentions and direct messages when you are offline or away from {siteName} for more than 5 minutes."
  footerId="user.settings.notifications.emailInfo"
  footerValues={
    Object {
      "siteName": "Mattermost",
    }
  }
  headerDefaultMessage="SEND EMAIL NOTIFICATIONS"
  headerId="mobile.notification_settings.email.send"
  theme={
    Object {
      "centerChannelBg": "#aaa",
      "centerChannelColor": "#aaa",
    }
  }
>
  <Component>
    <sectionItem
      action={[Function]}
      actionType="select"
      actionValue="30"
      label={
        <InjectIntl(FormattedText)
          defaultMessage="Immediately"
          id="user.settings.notifications.email.immediately"
        />
      }
      selected={true}
      theme={
        Object {
          "centerChannelBg": "#aaa",
          "centerChannelColor": "#aaa",
        }
      }
    />
    <Component
      style={
        Object {
          "backgroundColor": undefined,
          "flex": 1,
          "height": 1,
          "marginLeft": 15,
        }
      }
    />
    <sectionItem
      action={[Function]}
      actionType="select"
      actionValue="0"
      label={
        <InjectIntl(FormattedText)
          defaultMessage="Never"
          id="user.settings.notifications.email.never"
        />
      }
      selected={false}
      theme={
        Object {
          "centerChannelBg": "#aaa",
          "centerChannelColor": "#aaa",
        }
      }
    />
  </Component>
</section>
`;
